bpf/enhancedrecording/command.bpf.c

```diff
@@ -42,9 +44,20 @@ BPF_COUNTER(lost);
 
 static int __submit_arg(void *ptr, struct data_t *data)
 {
-    bpf_probe_read_user(data->argv, sizeof(data->argv), ptr);
-    if (bpf_ringbuf_output(&execve_events, data, sizeof(struct data_t), 0) != 0)
-        INCR_COUNTER(lost);
+    long read = 0;
+    long total_read = 0;
+    do {
```

Comment:
With this solution, it looks like we're doing a lot of unnecessary work here. Cannot we just extend the argv size to 4KB? Copying one long buffer sounds way faster than copying multiple small chunks. Potentially, we could even extend it to 32KB to cover the Linux max for the arg size, but I think that with 4KB, we should cover 99.9% of use cases and not see any performance degradation.

Reply:
I'm not sure if I have understood this correctly. We cannot exceed the maximum stack size of 512 bytes (increase the `ARGSIZE` value). If we utilize an eBPF map, we could conduct a single read, but we would still need to generate multiple events. Is that what you mean?